src/iris/bin/sender.py

```diff
@@ -782,6 +782,10 @@ def set_target_contact(message):
     # If we already have a destination set (eg incident tracking emails) no-op this
     if 'destination' in message:
         return True
+    if message['role'] == 'literal_target':
```

Comment:
Set `destination` outside of the sender, probably in rpc. No need to wait so long, and keeps the expand/unexpand logic out of the sender.